doc/weights.xml

```diff
@@ -122,3 +122,98 @@ gap> EdgeWeights(T);
     </Description>
     </ManSection>
 <#/GAPDoc>
+
+<#GAPDoc Label="EdgeWeightedDigraphShortestPaths">
+<ManSection>
+  <Attr Name="EdgeWeightedDigraphShortestPaths" Label="for a digraph" Arg="digraph"/>
+  <Oper Name="EdgeWeightedDigraphShortestPaths" Label="for a digraph and a pos int" Arg="digraph, source"/>
+  <Returns>A record.</Returns>
+  <Description>
+    If <A>digraph</A> is an edge-weighted digraph, this attribute returns a
+    record describing the paths of lowest total weight (the <E>shortest
+    paths</E>) connecting each pair of vertices.  If the optional argument
+    <A>source</A> is specified and is a vertex of <A>digraph</A>, then the
+    output will only contain information on paths originating from that
+    vertex. <P/>
+
+    In the two-argument form, the value returned is a record containing three
+    components: <C>distances</C>, <C>parents</C> and <C>edges</C>.  Each of
+    these is a list of integers with one entry for each vertex <C>v</C> as
+    follows: <P/>
+    <List>
+      <Item>
+        <C>distances[v]</C> is the total weight of the shortest path from
+        <A>source</A> to <C>v</C>.
+      </Item>
+      <Item>
+        <C>parents[v]</C> is the final vertex before <C>v</C> on the shortest
+        path from <A>source</A> to <C>v</C>.
+      </Item>
+      <Item>
+        <C>edges[v]</C> is the index of the edge of lowest weight going from
+        <C>parents[v]</C> to <C>v</C>.
+      </Item>
+    </List>
+    Using these three components together, you can find the shortest edge
+    weighted path to all other vertices from a starting vertex. <P/>
+
+    If no path exists from <A>source</A> to <C>v</C>, then <C>parents[v]</C> and
+    <C>edges[v]</C> will both be <K>fail</K>.  The distance from <A>source</A>
+    to itself is considered to be 0, and so both <C>parents[<A>source</A>]</C> and
+    <C>edges[<A>source</A>]</C> are <K>fail</K>.
+    Edge weights can have negative values, but this operation will fail with an
+    error if a negative-weighted cycle exists. <P/>
```

Comment:
Not sure that this is accurate, isn't it only the case that it'll fail when the input digraph is "dense", and Floyd-Warshall is applied?